libkbfs/keybase_service_base.go

```diff
@@ -574,6 +560,53 @@ func (k *KeybaseServiceBase) ResolveImplicitTeamByID(
 	return res.Name, nil
 }
 
+func (k *KeybaseServiceBase) checkForRevokedVerifyingKey(
+	ctx context.Context, currUserInfo UserInfo, kid keybase1.KID) (
+	newUserInfo UserInfo, exists bool, err error) {
+	k.log.CDebugf(ctx, "Checking merkle info for user %s, revoked key %s",
+		currUserInfo.UID, kid)
+
+	for key, info := range currUserInfo.RevokedVerifyingKeys {
+		if !key.KID().Equal(kid) {
```

Comment:
Would be nice if `RevokedVerifyingKeys` was indexed by `KID` so we could just look it up rather than linear searching. But not a big deal.

Reply:
The problem is `KBPKIClient` does the opposite thing, where it has the `VerifyingKey` and needs to do a lookup in `RevokedVerifyingKeys`.  So there's not a great solution without reworking the interface APIs.  Probably a project better left for some other time.

Reply:
Yeah agreed.